### PR TITLE
Separate adding award points from making payment

### PR DESCRIPTION
**Pull request by victornoel**
This is for #1162.

I wasn't exactly sure what is the proper way of continuing #1162, so I introduced a `@todo` but I'm open to improve it.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1740](https://github.com/zerocracy/farm/issues/1740) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1740](https://github.com/zerocracy/farm/issues/1740) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/2721553458); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy [hunk 1]
```diff
@@ -34,12 +34,21 @@ import com.zerocracy.pmo.banks.Payroll
...
   new Assume(project, xml).type('Make payment')
+  Farm farm = binding.variables.farm
   ClaimIn claim = new ClaimIn(xml)
+  /**
+   * @todo #1162:30min In continuation of #1162, add a test to ensure that a payment
+   *  results on points being awarded and continue the work so that the
+   *  API of "Make Payment" always receive a "cash" parameter on top of the minutes.
+   */
+  claim.copy()
+      .type('Add award points')
```
Comment (krzyk) [inline src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy]: @victornoel in groovy scripts we make 2 space indentations

File: src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy [hunk 2]
```diff
@@ -34,12 +34,21 @@ import com.zerocracy.pmo.banks.Payroll
 def exec(Project project, XML xml) {
   new Assume(project, xml).notPmo()
   new Assume(project, xml).type('Make payment')
+  Farm farm = binding.variables.farm
   ClaimIn claim = new ClaimIn(xml)
+  /**
+   * @todo #1162:30min In continuation of #1162, add a test to ensure that a payment
```
[the remark this answers is not in the document]
Comment (victornoel) [reply src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy]: @krzyk ok!
@g4s8 Could you advise on other things to add to this `@todo` to continue the work you outlined in https://github.com/zerocracy/farm/issues/1162#issuecomment-425318993

File: src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy (hunk 1 above)
Comment (victornoel) [reply src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy]: @krzyk sorry about that, it was an easy one ;)

File: src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy (hunk 2 above)
[the remark this answers is not in the document]
Comment (g4s8) [reply src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy]: @victornoel I'd add a link to this comment here

[the remark this answers is not in the document]
Comment (victornoel) [reply src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy]: @g4s8 ok, I will just put a summary of what we said then, no need to add an extra link with nothing informative in it.

**Comment by victornoel** [comment]
@krzyk I've pushed the changes requetsed

File: src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy [hunk 3]
```diff
@@ -34,12 +34,21 @@ import com.zerocracy.pmo.banks.Payroll
...
+   */
+  claim.copy()
+    .type('Add award points')
+    .postTo(new ClaimsOf(farm, project))
   String job = claim.param('job')
   String login = claim.param('login')
   String reason = new Par.ToText(claim.param('reason')).toString()
   int minutes = Integer.parseInt(claim.param('minutes'))
-  if (minutes < 0) {
+  if (minutes == 0) {
```
Comment (g4s8) [inline src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy]: @victornoel does this change required? Now someone can sent negative cash by mistake

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

File: src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy (hunk 3 above)
Comment (victornoel) [reply src/main/groovy/com/zerocracy/stk/pm/cost/make_payment.groovy]: @g4s8 yes, it's clearly not adequate, I reverted it!

**Comment by victornoel** [comment]
@g4s8 I took your comment into account, it should be good now

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1740](https://github.com/zerocracy/farm/issues/1740) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@krzyk  According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one major issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by krzyk** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@krzyk thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**